test/functional/feature_coinstatsindex.py

```diff
@@ -295,6 +300,53 @@ def _test_index_rejects_hash_serialized(self):
         for use_index in {True, False, None}:
             assert_raises_rpc_error(-8, msg, self.nodes[1].gettxoutsetinfo, hash_type='hash_serialized_2', hash_or_height=111, use_index=use_index)
 
+    def _test_dumpcoinstats_rpc(self):
+        self.log.info("Test dumpcoinstats RPC")
+
+        msg = "File name and path for dump file are required"
+        assert_raises_rpc_error(-8, msg, self.nodes[1].dumpcoinstats, None)
+
+        for cumulative in [True, False]:
+            if cumulative:
+                file = "ctest.csv"
+            else:
+                file = "test.csv"
+
+            path = os.path.join(get_datadir_path(self.options.tmpdir, 1), file)
```

Comment:
Probably time to introduce a helper and use that:

```diff
diff --git a/test/functional/test_framework/test_node.py b/test/functional/test_framework/test_node.py
index 7d2db391b6..aa9ff9831f 100755
--- a/test/functional/test_framework/test_node.py
+++ b/test/functional/test_framework/test_node.py
@@ -381,9 +381,13 @@ class TestNode():
     def wait_until_stopped(self, timeout=BITCOIND_PROC_WAIT_TIMEOUT):
         wait_until_helper(self.is_node_stopped, timeout=timeout, timeout_factor=self.timeout_factor)
 
+    @property
+    def datadir_path(self) -> Path:
+        return Path(self.datadir)
+
     @property
     def chain_path(self) -> Path:
-        return Path(self.datadir) / self.chain
+        return self.datadir_path / self.chain
 
     @property
     def debug_log_path(self) -> Path:
```

```suggestion
            path = self.nodes[1].datadir_path /  file
```

Reply:
I like the helper but using the `/` operator would mean `file` would also need to be a Path object iiuc. So, instead I opted for a slightly different approach that I liked a little better using another helper `datadir_file_path`. What do you think?

Reply:
> but using the / operator would mean file would also need to be a Path object iiuc

Does it? (It doesn't for me locally)

Reply:
Hm, seems like I misinterpreted the error, actually the next line fails. And I am now also seeing it fail with cast to `Path` but I am sure it previously passed at least twice with the cast. Very strange. Maybe a python version or platform issue, need to look deeper into this.

```
2022-06-11T16:28:24.998000Z TestFramework (INFO): Test dumpcoinstats RPC
2022-06-11T16:28:24.998000Z TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/Users/FJ/projects/clones/bitcoin/test/functional/test_framework/test_framework.py", line 133, in main
    self.run_test()
  File "/Users/FJ/projects/clones/bitcoin/test/functional/feature_coinstatsindex.py", line 58, in run_test
    self._test_dumpcoinstats_rpc()
  File "/Users/FJ/projects/clones/bitcoin/test/functional/feature_coinstatsindex.py", line 319, in _test_dumpcoinstats_rpc
    self.nodes[1].dumpcoinstats(path, cumulative)
  File "/Users/FJ/projects/clones/bitcoin/test/functional/test_framework/coverage.py", line 49, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/Users/FJ/projects/clones/bitcoin/test/functional/test_framework/authproxy.py", line 141, in __call__
    postdata = json.dumps(self.get_request(*args, **argsn), default=EncodeDecimal, ensure_ascii=self.ensure_ascii)
  File "/Users/FJ/projects/clones/bitcoin/test/functional/test_framework/authproxy.py", line 131, in get_request
    json.dumps(args or argsn, default=EncodeDecimal, ensure_ascii=self.ensure_ascii),
  File "/opt/homebrew/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 234, in dumps
    return cls(
  File "/opt/homebrew/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/opt/homebrew/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Users/FJ/projects/clones/bitcoin/test/functional/test_framework/authproxy.py", line 66, in EncodeDecimal
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: PosixPath('/var/folders/9z/n7rz_6cj3bq__11k5kcrsvvm0000gn/T/bitcoin_func_test_ru27w0qw/node1/ctest.csv') is not JSON serializable
```

Reply:
You could do `str(path)` before passing it as json?

Reply:
Yeah, that works too and I just pushed a version with that. But I find it a little less elegant overall.